catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -0,0 +1,507 @@
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.measure.Distance;
+
+public class SolrMetacardClientImpl implements SolrMetacardClient {
+
+    protected static final String RELEVANCE_SORT_FIELD = "score";
+
+    private static final String DISTANCE_SORT_FUNCTION = "geodist()";
+
+    private static final String DISTANCE_SORT_FIELD = "_distance_";
+
+    private static final String GEOMETRY_SORT_FIELD =
+            Metacard.GEOGRAPHY + SchemaFields.GEO_SUFFIX + SchemaFields.SORT_KEY_SUFFIX;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SolrMetacardClientImpl.class);
+
+    private static final String QUOTE = "\"";
+
+    private final SolrClient client;
+
+    private final SolrFilterDelegateFactory filterDelegateFactory;
+
+    private final FilterAdapter filterAdapter;
+
+    private final DynamicSchemaResolver resolver;
+
+    public SolrMetacardClientImpl(SolrClient client, FilterAdapter catalogFilterAdapter,
+            SolrFilterDelegateFactory solrFilterDelegateFactory,
+            DynamicSchemaResolver dynamicSchemaResolver) {
+        this.client = client;
+        filterDelegateFactory = solrFilterDelegateFactory;
+        filterAdapter = catalogFilterAdapter;
+        resolver = dynamicSchemaResolver;
+    }
+
+    public SolrClient getClient() {
+        return client;
+    }
+
+    @Override
+    public SourceResponse query(QueryRequest request) throws UnsupportedQueryException {
+        if (request == null || request.getQuery() == null) {
+            return new QueryResponseImpl(request, new ArrayList<Result>(), true, 0L);
+        }
+
+        SolrQuery query = getSolrQuery(request, filterDelegateFactory.newInstance(resolver));
+
+        long totalHits;
+        List<Result> results = new ArrayList<>();
+        try {
+            QueryResponse solrResponse = client.query(query, SolrRequest.METHOD.POST);
+            totalHits = solrResponse.getResults()
+                    .getNumFound();
+            SolrDocumentList docs = solrResponse.getResults();
+
+            for (SolrDocument doc : docs) {
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug("SOLR DOC: {}",
+                            doc.getFieldValue(Metacard.ID + SchemaFields.TEXT_SUFFIX));
+                }
+                ResultImpl tmpResult;
+                try {
+                    tmpResult = createResult(doc);
+                    // TODO: register metacard type???
+                } catch (MetacardCreationException e) {
+                    throw new UnsupportedQueryException("Could not create metacard(s).", e);
+                }
+
+                results.add(tmpResult);
+            }
+
+        } catch (SolrServerException | IOException | SolrException e) {
+            throw new UnsupportedQueryException("Could not complete solr query.", e);
+        }
+
+        SourceResponseImpl sourceResponseImpl = new SourceResponseImpl(request, results);
```

Comment:
`SourceResponse` does not have the `setHits` method that is used at 139. But there is a constructor that can also take total hits which could be used instead.